nixos/modules/virtualisation/oci-containers.nix

```diff
@@ -262,13 +309,79 @@ let
         };
 
         autoStart = mkOption {
-          type = types.bool;
+          type = with types; bool;
           default = true;
           description = ''
             When enabled, the container is automatically started on boot.
             If this option is set to false, the container has to be started on-demand via its service.
           '';
         };
+
+        pull = mkOption {
+          type = with types; enum ["always" "missing" "never" "newer"];
+          default = "missing";
+          description = ''
+            Image pull policy for the container. Must be one of: always, missing, never, newer
+          '';
+        };
+
+        capAdd = mkOption {
+          type = with types; listOf (enum capabilities);
+          default = [];
+          description = ''
+            Capabilities to add to container
+          '';
+          example = literalExpression ''
+            [
+              SYS_ADMIN
+            ]
+          '';
+        };
+
+        capDrop = mkOption {
+          type = with types; listOf (enum capabilities);
+          default = [];
+          description = ''
+            Capabilities to drop from container
+          '';
+          example = literalExpression ''
+            [
+              SYS_ADMIN
+            ]
+          '';
+        };
```

Comment:
A type that's nicer to use instead of these two options would be `lazyAttrsOf (nullOr bool)`. Now that doesn't have built-in checking of the attribute names, but it does allow for better control of overriding, allowing `mkDefault true` and `mkForce false` for example.

That's [what arion does](https://docs.hercules-ci.com/arion/options#_services_name_service_capabilities) for example, but it doesn't check the names; I think docker compose checks it later.
You could check the attribute names by adding to `assertions`.

A parameter for checking the names would be a nice addition to [`attrsWith`](https://github.com/NixOS/nixpkgs/pull/344216) when that lands.

Reply:
I've extended the existing test to cover new config options. I see arion doesn't assert capabilities names either perhaps it's not necessary given a nonexistent capability will cause the service to fail to start